Sources/UID2/UID2Manager.swift

```diff
@@ -128,12 +137,37 @@ public final actor UID2Manager {
     
     // MARK: - Internal Identity Lifecycle
     
+    private func setIdentityPackage(_ identity: IdentityPackage) async {
+        if let _ = await validateAndSetIdentity(identity: identity.identity,
+                                                status: identity.status,
+                                                statusText: identity.errorMessage) {
+            
+            // An identity's status can change based upon the current time and it's expiration. We will schedule some work
+            // to detect when it changes so that we can report it accordingly.
+            await checkIdentityExpiration()
+
+            // After a new identity has been set, we have to work out how we're going to potentially refresh it. If the
+            // identity is null, because it's been reset of the identity has opted out, we don't need to do anything.
+            await checkIdentityRefresh()
+        }
+    }
+    
+    private func loadStateFromDisk() async {
+        if let identity = KeychainManager.shared.getIdentityFromKeychain() {
+            // Has Opted Out?
+            //  - Handled by setIdentityPackage() validateAndSetIdentity()
+            // Has Identity Token Expired with valid RefreshToken?
+            //  - Handled by setIdentityPackage() triggerRefreshOrSetTimer()
+            await setIdentityPackage(identity)
```

Comment:
I love this, exactly what I did on Android. It means that we immediately detect expiration (or refresh) after restoring from persistence.